src/proxy.c

```diff
@@ -54,6 +54,12 @@ int proxy_process(ProxyRequest *request, ProxyReply *reply)
         case P_GET_BOOTARGS:
             reply->retval = boot_args_addr;
             break;
+        case P_GET_BOOT_CPU_IDX:
+            reply->retval = boot_cpu_idx;
```

Comment:
I don't think it's a good idea to retrieve something which is only valid after `smp_start_secondaries` is called.

Reply:
Thoughts on how to approach this?
This was all sort of a hack to get it working.
I could make smp_start_secondaries return this value as an out arg or something?
Or some totally different approach?

Reply:
I would use the `state` property from the ADT. In two cases it is iterating over all CPUs and skipping the boot CPU. The remaining use can be replaced with `[cpu for cpu in self.u.adt["cpus"] if cpu.state == "running"]`.